ceph_s3tests_runner.py

```diff
@@ -60,19 +76,158 @@ def get_tests_from_te(jira_obj, args, test_type=None):
         test_type = ['ALL']
     test_list, _ = jira_obj.get_test_ids_from_te(str(args.te_ticket), test_type)
     if len(test_list) == 0:
-        raise EnvironmentError("Please check TE provided, tests or tag is missing")
+        raise EnvironmentError("Could not find tests matching given test type.")
     return test_list
 
 
 def collect_test_info(jira_obj, test):
     """Collect Test information."""
     test_details = jira_obj.get_issue_details(test)
-    test_name = test_details.fields.summary
     test_to_run = test_details.fields.customfield_20984
-    test_label = ''
-    if test_details.fields.labels:
-        test_label = test_details.fields.labels[0]
-    return test_name, test_label, test_to_run
+    if not test_to_run:
+        raise EnvironmentError(f"Definition field for {test} JIRA is empty.")
+    return test_to_run
+
+
+def setup_configurations(args):
+    """Setup host:port and security protocol"""
+    setup_details = ClientConfig(get_db_credential()).get_setup_details(args.target)
+    if not os.path.exists(CONFIG_FILE):
+        LOGGER.info("Coping default config file")
+        shutil.copyfile("scripts/ceph_s3_tests/cortx_rgw_template.conf", CONFIG_FILE)
+    config = ConfigParser()
+    config.read(CONFIG_FILE)
+    LOGGER.info({section: dict(config[section]) for section in config.sections()})
+    config.set('DEFAULT', 'host', str(setup_details['lb'].split(":")[0]))
+    config.set('DEFAULT', 'port', str(setup_details['lb'].split(":")[1]))
+    config.set('DEFAULT', 'is_secure', str(args.use_ssl))
+    config.set('DEFAULT', 'ssl_verify', str(args.validate_certs))
+    with open(CONFIG_FILE, 'w') as configfile:
+        config.write(configfile)
+
+
+def check_or_create_accounts():
+    """Create IAM accounts required for ceph test execution if accounts does not exit already"""
+    iam_users_sections = ["s3 main", "s3 alt", "s3 tenant", "iam"]
+    csm_obj = csm_api_factory("rest")
+
+    resp = csm_obj.list_iam_users_rgw()
+    if resp.status_code != HTTPStatus.OK:

```

Comment:
will this give u error even if no IAM users are created?

Reply:
Even if no users, response status code should be 200 (OK)